cloudevents/extensions/deprecation.md

```diff
@@ -0,0 +1,58 @@
+# Deprecation and Sunset extension
+
+This specification defines attributes that can be included in CloudEvents to
+indicate to [consumers](../spec.md#consumer) or
+[intermediaries](../spec.md#intermediary) the deprecation and sunset status of
+the event type. These attributes inform consumers of CloudEvents about upcoming
+changes or removals, facilitating smoother transitions and proactive
+adjustments.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean it
+needs to be in all CloudEvents, rather it needs to be included only when this
+extension is being used.
+
+## Attributes
+
+### deprecation
+
+- Type: `String`
+- Description: Indicates that the event type is deprecated. This can be a
+  boolean `true` or a date in [RFC 5322](https://tools.ietf.org/html/rfc5322)
```

Comment:
Rather than having a String that can be one of two formats, I'd prefer to have two attributes, each of which uses a single existing CE type - one `Boolean`, one `Timestamp`. (e.g. `deprecated` and `deprecatedfrom`)

Reply:
separated out the attributes:
- deprecation
- deprecationfrom